### DISCUSSION TITLE
How I can store the Nested json list in a local data base and access them using moor?

### DISCUSSION THREAD

**Discussion by DhavalRKansara** [simolus3/drift#1374] [Q&A]
My JSON is as below:

```
{
  "results": [
    {
      "gender": "male",
      "name": {
        "title": "Mr",
        "first": "Vernon",
        "last": "Lane"
      },
      "email": "vernon.lane@example.com",
      "phone": "(646)-709-6968",
      "cell": "(908)-582-2920",
      "picture": {
        "large": "https://randomuser.me/api/portraits/men/70.jpg",
        "medium": "https://randomuser.me/api/portraits/med/men/70.jpg",
        "thumbnail": "https://randomuser.me/api/portraits/thumb/men/70.jpg"
      }
    },
    {
      "gender": "male",
      "name": {
        "title": "Mr",
        "first": "Vernon",
        "last": "Lane"
      },
      "email": "vernon.lane@example.com",
      "phone": "(646)-709-6968",
      "cell": "(908)-582-2920",
      "picture": {
        "large": "https://randomuser.me/api/portraits/men/70.jpg",
        "medium": "https://randomuser.me/api/portraits/med/men/70.jpg",
        "thumbnail": "https://randomuser.me/api/portraits/thumb/men/70.jpg"
      }
    }
  ]
}
```

**Comment by simolus3** [comment]
I think the best approach to store this information is to use [type converters](https://moor.simonbinder.eu/docs/advanced-features/type_converters/). By using other packages like `built_value` or `json_serializable`, you can write a class representing an element in that array (so it would have gender, name, email, ...) properties that can be mapped from and to json. Your type converter would then map that array to a string so that it can be stored in the database.

Depending on your use case, you can also extract some information into additional tables. For instance, you could choose to only store the picture and the name as a serialized string and write a table for the rest:

```dart
class Users extends Table {
  TextColumn get gender => text()();
  TextColumn get email => text()();
  // ...
}
```

In general, an advantage of type converters is that they're easier to use. Especially for deeply-nested data and arrays, SQL can be a bit of a pain for simple queries. On the other hand. extracting information into tables enables more efficient queries. It also allows you to select a subset of data more easily.
